### ISSUE TITLE
Attach Invoice With Spaces in Filename Causes 404 Error - Xero.NetStandard.OAuth2 9.2.0

### ISSUE THREAD

**Issue by phil000** [XeroAPI/Xero-NetStandard#570]
**SDK you're using (please complete the following information):**
 - Version Xero.NetStandard.OAuth2 9.2.0

**Describe the bug**
We have upgraded from Xero.NetStandard.OAuth2 6.0.0 to **Xero.NetStandard.OAuth2 9.2.0**
When uploading files to attach to an invoice we now get an error code 404, with error content "The resource you're looking for cannot be found"
If we remove the spaces from the filename the upload works correctly.
We have noticed that this error was introduced in **Xero.NetStandard.OAuth2 9.0.0**
The issue does not occur in 8.1.0.

**To Reproduce**
Upload a file with spaces.

![image](https://github.com/user-attachments/assets/c70fb8e3-c636-4643-b4b4-5a048f33a450)

**Expected behavior**
File attachment with spaces can be uploaded via .NET API.

**Comment by phil000** [comment]
We have noted that while the documentation states:
_"When the file name includes special characters, the characters should not be encoded unless they're brackets. Brackets must be encoded in order for the call to go through, and all other characters must be unencoded"_

If we actually  URL path encode the filename the filename with spaces can be uploaded again. (This means replacing spaces with %20, not +)

**Comment by DanPatten** [comment]
This just happened to me, I had to use System.Uri.EscapeDataString instead of  WebUtility.UrlEncode

**Comment by phil000** [comment]
Yes, FYI the answer is to use **System.Uri.EscapeDataString** that escapes spaces as "%20"

System.Net.WebUtility.UrlEncode encodes spaces as '+' which does not work.

**Comment by joshsj** [comment]
> Yes, FYI the answer is to use **System.Uri.EscapeDataString** that escapes spaces as "%20"
> 
> System.Net.WebUtility.UrlEncode encodes spaces as '+' which does not work.

+1, this is what RestSharp is using (and therefore what this SDK was using up to 9.x.x).

For reference:

- [Where the SDK created the `RestClientOptions`](https://github.com/XeroAPI/Xero-NetStandard/blob/64be5ee6acda8bc033db6dfe08c1e1dcda658916/Xero.NetStandard.OAuth2/Client/ApiClient.cs#L502-L505). No encoder is specified so it uses the default.

- [The default encoder](https://github.com/restsharp/RestSharp/blob/bf4008f972257908d666b2884ec7fe330b759424/src/RestSharp/Options/RestClientOptions.cs#L227-L230) uses an extension method internal to RestSharp.

- [This extension method](https://github.com/restsharp/RestSharp/blob/bf4008f972257908d666b2884ec7fe330b759424/src/RestSharp/Extensions/StringExtensions.cs#L41-L69) that uses `Uri.EscapeDataString`. It also handles unencoded URLs longer than 32766 characters.

This raises the question of whether compatibility with massive URLs should also be preserved as part of the fix? Seems like the sensible option given we can simply adopt RestSharp's method - there's a [simplified version](https://github.com/XeroAPI/Xero-NetStandard/blob/4b34a22312d280a9e2b125610a9ec9b8216f7f3d/Xero.NetStandard.OAuth2/Client/ClientUtils.cs#L121-L154) in the codebase already.

**Comment by phil000** [comment]
Anything happening here? This seems like a pretty major and unnecessary breaking change. Please un-break ASAP.
